common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -60,18 +63,26 @@ public void recycle(Object object) {
         if (maxCapacity < 0) {
             maxCapacity = DEFAULT_INITIAL_MAX_CAPACITY;
         }
-
         DEFAULT_MAX_CAPACITY = maxCapacity;
 
+        int maxSharedCapacityFactor = SystemPropertyUtil.getInt("io.netty.recycler.maxSharedCapacityFactor",
+                DEFAULT_MAX_SHARED_CAPACITY_FACTOR);
+        if (maxSharedCapacityFactor < 0) {
+            maxSharedCapacityFactor = DEFAULT_MAX_SHARED_CAPACITY_FACTOR;
+        }
+        MAX_SHARED_CAPACITY_FACTOR = maxSharedCapacityFactor;
```

Comment:
also `0` is not really "valid" in this context bcz it will result in divide by 0. this condition is caught below in the constructor ... but maybe more clear just to disallow it here?